akka-docs/src/test/scala/docs/future/FutureDocSpec.scala

```diff
@@ -479,31 +482,37 @@ class FutureDocSpec extends AkkaSpec {
   }
 
   "demonstrate usage of pattern.after" in {
+    import akka.actor.typed.scaladsl.adapter.ClassicActorSystemOps
+    implicit val system: ActorSystem[Nothing] = this.system.toTyped
     //#after
-    // TODO after is unfortunately shadowed by ScalaTest, fix as part of #3759
-    // import akka.pattern.after
-
     val delayed =
-      akka.pattern.after(200 millis, using = system.scheduler)(Future.failed(new IllegalStateException("OHNOES")))
+      akka.pattern.after(200.millis)(Future.failed(new IllegalStateException("OHNOES")))
+
     val future = Future { Thread.sleep(1000); "foo" }
     val result = Future.firstCompletedOf(Seq(future, delayed))
     //#after
     intercept[IllegalStateException] { Await.result(result, 2 second) }
   }
 
   "demonstrate pattern.retry" in {
+    import akka.actor.typed.scaladsl.adapter.ClassicActorSystemOps
+    val system: ActorSystem[Nothing] = this.system.toTyped
     //#retry
-    implicit val scheduler = system.scheduler
+    import akka.actor.typed.scaladsl.adapter._
+    implicit val scheduler: akka.actor.Scheduler = system.scheduler.toClassic
```

Comment:
How would we like to offer a method that takes a `ClassicActorSystemProvider` instead? 
Use a new name?

Reply:
maybe an implicit conversion could work for Scala, see `akka.actor.typed.scaladsl.AskPattern.schedulerFromActorSystem`, which makes it possible to use `ask` with an `implicit ActorSystem` instead of an `implicit Scheduler`

Reply:
This lives in akka-actor. That would mean to go all-in on this and add a new APIs implementation, I guess?